operator/pkg/controllers/etcd/statefulset.go

```diff
@@ -32,10 +32,17 @@ func (c *Controller) reconcileStatefulSet(ctx context.Context, controlPlane *v1a
 	// Generate the default pod spec for the given control plane, if user has
 	// provided custom config for the etcd pod spec, patch this user
 	// provided config to the default spec
-	etcdSpec, err := patch.PodSpec(podSpecFor(controlPlane), controlPlane.Spec.Etcd.Spec)
+	podSpec, err := patch.PodSpec(podSpecFor(controlPlane), controlPlane.Spec.Etcd.Spec)
 	if err != nil {
 		return fmt.Errorf("failed to patch pod spec, %w", err)
 	}
+	// Generate the default volume claim template spec for the given control plane, if user has
+	// provided custom config for the etcd volume template spec, patch this user
+	// provided config to the default spec
+	volumeClaimSpec, err := patch.VolumeClaimTemplateSpec(DefaultVolumeClaimTemplateSpec(), controlPlane.Spec.Etcd.VolumeClaimTemplates)
+	if err != nil {
+		return fmt.Errorf("failed to patch voluemclaim template spec, %w", err)
```

Comment:
We know what these types are and control the input/output. Can we write them in a way that doesn't require expressive error handling? We can instead panic, which will only happen due to a programmer error.